content/4_Gold/LIS.mdx

```diff
@@ -338,6 +342,68 @@ class PCB{
 
 </LanguageSection>
 
+## Extension - LCS on Permutations
+
+Given two permutations of length $N$ ($1\le N \le 10^5$), find the longest common subsequence.
+
+### Slow Solution
+
+**Time Complexity**: $\mathcal O(N^2)$
+
+See [here](/gold/paths-grids#solution---longest-common-subsequence).
+
+### Efficient Solution
+
+**Time Complexity**: $\mathcal O(N\log N)$
+
+In the $\mathcal O(N^2)$ solution, we increase the DP arrray only if $a[i] = b[j]$, but since both arrays are permutations of length $N$, for each $a[i]$, there must be only one matching element in $b$.  
+
+Let us create an array $c$ where $c[i]$ is $a^{-1}[b[i]]$. Our final LIS $c[x_1], c[x_2], \dots c[x_k]$ corresponds to the LCS $b[x_1], b[x_2], \dots b[x_k]$, which is equivalent to $a[c[x_1]], a[c[x_2]], \dots a[c[x_k]]$.
+
+In other words, every increasing subsequence $x_{1 \dots k}$ in $c$ corresponds to a common subsequence between $a$ and $b$.
```

Comment:
```suggestion
Let us create an array $c$ where $c[i]$ is $a^{-1}[b[i]]$ ($a^{-1}$ is the inverse of $a$: $a^{-1}[x]$ is the index where element $x$ is located. Specifically $a[a^{-1}[x]]=x and a^{-1}[a[i]]=i$). 

Notice that every increasing subsequence $x_{1 \dots k}$ in $c$ corresponds to a common subsequence between $a$ and $b$. Increasing subsequence $c[x_1], c[x_2], \dots c[x_k]$ corresponds to the the common subsequence $b[x_1], b[x_2], \dots b[x_k]$, which is equivalent to $a[c[x_1]], a[c[x_2]], \dots a[c[x_k]]$. Thus, the length of the longest common subsequence between $a$ and $b$ is the longest increasing subsequence of $c$.
```

Something like this probably flows better (my initial suggestion was more of some stuff that would be good to have as opposed to structured info). Also, I notice there's leading spaces - I didn't add them but format it however you like.

I made it a bit wordy, maybe the thing explaining a^-1 is a bit messy/unclear... modify it how you like

Reply:
Or you could just define pos the way you did before instead of using the a^-1 notation